### DISCUSSION TITLE
I2S WS Pin in Mono Mode

### DISCUSSION THREAD

**Discussion by mrburns42** [micropython/micropython#9284] [Libraries & Drivers] [answered]
I have a SPM1423 MEMs microphone connected to an ESP32.    While my code builds and runs, the captured audio is just noise.   One difficulty is that the microphone assembly I am using does not bring out the WS pin.   Instead that pin is internally grounded within the assembly.    I am setting the microphone mode to MONO.    I assigned the WS pin to an unused pin on the ESP32 just to make the driver happy.   
What I suspect is that it is in MONO mode for the wrong channel.  However,  I do not see a way to select MONO right channel or MONO left channel.    Is there a way to do that?   Am I supposed to tie the WS pin low to signal to the driver which MONO channel?   

The ESP32 board is the M5 Stamp PICO.    The microphone is the PDM MEMs Microphone.  

Thanks.

**Comment by miketeachman** [comment, marked as answer]
The SPM1423 Microphone uses a PDM (Pulse Density Modulation) signal to transfer audio sample data and will not work with the I2S implementation in MicroPython.  Unfortunately, there is nothing you can do with this microphone to make it work with the current release of MicroPython.

**Reply by mrburns42** [reply]
OK,  Thank you for the super quick response.   I will have to go back to an Arduino implementation to use this HW.

**Reply by miketeachman** [reply]
You could also check out CircuitPython which is just starting to offer ESP32 support.  The `audiobusio` module supports PDM microphone using the class `audiobusio.PDMIn`.
https://circuitpython.org/board/adafruit_feather_huzzah32/
